monkestation/code/modules/machining/machining_exclusives.dm

```diff
@@ -0,0 +1,229 @@
+	if(src.loc == user)
+		if(!armed)
+			if(user.get_item_by_slot(ITEM_SLOT_BACK) == src)
+				armed = TRUE
+				if(!user.put_in_hands(gun))
+					armed = FALSE
+					to_chat(user, span_warning("You need a free hand to hold the gun!"))
+					return
+				update_appearance()
+				user.update_worn_back()
+		else
+			to_chat(user, span_warning("You are already holding the gun!"))
+	else
+		..()
+
+/obj/item/pulsepack/attackby(obj/item/W, mob/user, params)
+	if(W == gun) //Don't need armed check, because if you have the gun assume its armed.
+		user.dropItemToGround(gun, TRUE)
+	else
+		..()
+
+
+/obj/item/pulsepack/dropped(mob/user)
+	. = ..()
+	if(armed)
+		user.dropItemToGround(gun, TRUE)
+
+/obj/item/pulsepack/MouseDrop(atom/over_object)
+	. = ..()
+	if(armed)
+		return
+	if(iscarbon(usr))
+		var/mob/M = usr
+
+		if(!over_object)
+			return
+
+		if(!M.incapacitated())
+
+			if(istype(over_object, /atom/movable/screen/inventory/hand))
+				var/atom/movable/screen/inventory/hand/H = over_object
+				M.putItemFromInventoryInHandIfPossible(src, H.held_index)
+
+
+/obj/item/pulsepack/update_icon_state()
+	icon_state = armed ? "notholsteredp" : "holsteredp"
+	return ..()
+
+/obj/item/pulsepack/proc/attach_gun(mob/user)
+	if(!gun)
+		gun = new(src)
+	gun.forceMove(src)
+	armed = FALSE
+	if(user)
+		to_chat(user, span_notice("You attach the [gun.name] to the [name]."))
+	else
+		src.visible_message(span_warning("The [gun.name] snaps back onto the [name]!"))
+	update_appearance()
+	user.update_worn_back()
+
+
+/obj/item/gun/energy/pulse/makeshift
+	name = "Pulse caster"
+	desc = "More a set of aiming lenses, focuses the power generated from the pulse generator into deadly beams."
+	icon = 'icons/obj/weapons/guns/minigun.dmi'
+	icon_state = "pulsecaster"
+	inhand_icon_state = "pulsecaster"
+	lefthand_file = 'monkestation/icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'monkestation/icons/mob/inhands/weapons/guns_righthand.dmi'
+	slowdown = 1
+	slot_flags = null
+	w_class = WEIGHT_CLASS_HUGE
+	custom_materials = null
+	weapon_weight = WEAPON_HEAVY
+	ammo_type = list(/obj/item/ammo_casing/energy/laser/pulse)
+	cell_type = /obj/item/stock_parts/cell/pulsepack
+	item_flags = SLOWS_WHILE_IN_HAND
+	can_charge = FALSE
+	var/obj/item/pulsepack/ammo_pack
+
+/obj/item/gun/energy/pulse/makeshift/Initialize(mapload)
+	if(!istype(loc, /obj/item/pulsepack)) //We should spawn inside an ammo pack so let's use that one.
+		return INITIALIZE_HINT_QDEL //No pack, no gun
+	ammo_pack = loc
+	AddElement(/datum/element/update_icon_blocker)
+	AddComponent(/datum/component/automatic_fire, 0.2 SECONDS)
+	return ..()
+
+/obj/item/gun/energy/pulse/makeshift/Destroy()
+	if(!QDELETED(ammo_pack))
+		qdel(ammo_pack)
+	ammo_pack = null
+	return ..()
+
+/obj/item/gun/energy/pulse/makeshift/attack_self(mob/living/user)
+	return
+
+/obj/item/gun/energy/pulse/makeshift/dropped(mob/user)
+	SHOULD_CALL_PARENT(FALSE)
```

Comment:
Dont.

Just replace with `return ..()` at the end of the proc
This allows you to early return with the snowflake you want, but still allows dropped to function how its supposed to.